_episodes/02-file_parsing.md

```diff
@@ -10,22 +10,102 @@ objectives:
+{: .callout}
+
+## Reading a file
+
+In Python, there are many ways in python to read in information from a text file.  The best method to use depends on the type of data and the type of analysis you are performing.  If you have a file with lots of different types of information, text and numbers, with different types of formatting, the most generic way to read in information is the `readlines()` function.  Before you can read in a file, you have to open the file using the file path we defined above. This will create a file object, or filehandle.  The file we will be analyzing in this example is a [PSI4](http://www.psicode.org) output file for a SCF/cc-pVDZ energy calculation for an ethanol molecule.
 
 ```
-outfile = open("ethanol.out","r")
-data=outfile.readlines()
+outfile = open(ethanol_file,"r")
+data = outfile.readlines()
 ```
 {: .language-python}
-This code opens a file for reading (that's the `r` part in the `open` function), assigns it to the filehandle `outfile` and then uses the `readlines()` function.  Notice the dot notation introduced last lesson; readlines acts on the filehandle given right before the dot.  The function creates a list called data where each element of the list is a string that is one line of the file.  This is always how the `readlines()` function works.  
+This code opens a file for reading and assigns it to the filehandle  `outfile`. The `r` argument in the function stands for `read`. Other arguments might be `w` for `write` if we want to write new information to the file, or `a` for append if we want to add new information at the end of the file.
+
+In the next line, we use the `readlines` function to get our file as a list of strings.  Notice the dot notation introduced last lesson; readlines acts on the file object given right before the dot.  The function creates a list called data where each element of the list is a string that is one line of the file.  This is always how the
+`readlines()` function works.
+
+> ## `readlines` function behavior
+> Note that the `readlines` function can only be used on a file object one time. When you open the file for reading, a 'file pointer' points to the first line. Every time the `readlines` function finds a new line, this pointer is moved. When it finishes, the file pointer is at the end of the file. Thus, if you called readlines again, the pointer for the file object is already at the end of the file. If you forget to set `outfile.readlines()` equal to a variable, you must `open` the file again in order to get the contents of the file.
```

Comment:
This callout seems confusing to me because I don't think the students will understand what a file point is or why it is important.  If the take away is just that if you want to use readlines() again that you need to give it a new name, maybe we should just say that?

Reply:
Yes. I was just trying to emphasize that if they wanted to run `readlines` again on the same file, they would need to open it again. I'll try to phrase this on a more understandable way.